File: server/order-service/src/main/java/com/sudo248/orderservice/repository/entity/order/OrderStatus.java
package com.sudo248.orderservice.repository.entity.order;

import java.util.ArrayList;
import java.util.List;

public enum OrderStatus {
    // người bán hàng chuẩn bị đơn hàng.
    PREPARE,
    // Người bán hàng sẵn sàng giao cho shipper.
    READY,
    // Shipper đã nhận hàng.
    TAKE_ORDER,
    // Đang vận chuyển đơn hàng.
    SHIPPING,
    // Shipper đã giao hàng đến khách hàng
    DELIVERED,
    // Khách hàng đã nhận được hàng
    RECEIVED,
    // Đơn bị huỷ
    CANCELED;

    public static OrderStatus fromValue(String value) {
        if (value == null) return null;
        switch (value.toLowerCase()) {
            case "prepare":
                return PREPARE;
            case "ready":
                return READY;
            case "take_order":
                return TAKE_ORDER;
            case "shipping":
                return SHIPPING;
            case "delivered":
                return DELIVERED;
            case "received":
                return RECEIVED;
            case "canceled":
                return CANCELED;
            default:
                return null;
        }
    }

    public static List<OrderStatus> fromValues(String values) {
        if (values == null || values.isBlank())  return null;
        String[] listValues = values.split(",");
        List<OrderStatus> result = new ArrayList<>();
        for(String value : listValues) {
            OrderStatus status = fromValue(value.trim());
            if (status != null) {
                result.add(status);
            }
        }
        return result;
    }
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/order/OrderStatus.kt
package com.sudo248.sudoo.domain.entity.order

import androidx.annotation.StringRes
import com.sudo248.sudoo.R

enum class OrderStatus(val value: String, @StringRes val displayValue: Int) {
    PREPARE("prepare", R.string.prepare),
    READY("ready", R.string.ready),
    TAKE_ORDER("take_order", R.string.take_order),
    SHIPPING("shipping", R.string.shipping),
    DELIVERED("delivered", R.string.delivered),
    RECEIVED("received", R.string.received),
    CANCELED("canceled", R.string.canceled);
}